compendium/DeclarativeServices/test/TestComponentLifecycle.cpp

```diff
@@ -175,13 +187,82 @@ TEST_F(tServiceComponent, testImmediateComponent_LifeCycle_Dynamic) // DS_TOI_51
   EXPECT_FALSE(static_cast<bool>(sRef2)) << "Service must not be available after it's dependency is removed";
 }
 
+//Function to validate lazy loading of delayed component
+void lazyLoadingValidation(std::string expectedResult)
+{
+  #if defined(_WIN32)
+
+    HMODULE hMods[1024];
+    DWORD cbNeeded;
+    unsigned int i;
+
+    HANDLE hProcess = GetCurrentProcess();
+    auto res = EnumProcessModules(hProcess, hMods, sizeof(hMods), &cbNeeded);
+    EXPECT_GT(sizeof(hMods), cbNeeded) << "Size of array is too small to hold all module handles";
+    EXPECT_EQ(res, 1) << "Enumeration failed";
+
+    TCHAR szModName[MAX_PATH];
+    std::string result;
+
+    for (i = 0; i < (cbNeeded / sizeof(HMODULE)); i++)
+    {
+        auto file = GetModuleFileNameA(hMods[i], szModName, sizeof(szModName) / sizeof(TCHAR));
+        EXPECT_NE(file, 0) << "Failed to retrive file path";
+        result += szModName;
+    }
+
+    std::size_t found = result.find("TestBundleDSTOI6");
```

Comment:
some duplicate code here. Also appears in the else. Might be worth pulling it into a function?